### PR TITLE
test(testutil): add new function for creating a test bucket

### PR DESCRIPTION
**Pull request by BrennaEpp**
Part 1 to bucket creation cleanup.
Part 2 will be a PR replacing all references to CleanBucket with TestBucket or CreateTestBucket.
Part 3 will remove the CleanBucket function.

Deprecates CleanBucket; a followup PR will remove all references to this function. CleanBucket can run into problems of availability due to fast creation after deletion. Instead, unique randomly generated bucket names should be used.

To replace CleanBucket, TestBucket provides a simple way to create a bucket for testing, creating this random name and taking charge of the cleanup.

## Checklist
- [x] I have followed [Contributing Guidelines from CONTRIBUTING.MD](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md)
- [x] **Tests** pass:   `go test -v ./..` (see [Testing](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#testing))
- [x] **Code formatted**:   `gofmt` (see [Formatting](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#formatting))
- [x] **Vetting** pass:   `go vet` (see [Formatting](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#formatting))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [ ] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: internal/testutil/storage_cleaner.go [hunk 1]
```diff
@@ -28,15 +28,43 @@ import (
...
+		if err := DeleteBucketIfExists(ctx, client, bucketName); err != nil {
+			log.Printf("Bucket.Delete(%q): %v", bucketName, err)
+		}
+	})
+	return bucketName
 }
 
 // CleanBucket creates a new bucket. If the bucket already exists, it will be
 // deleted and recreated.
+// Deprecated: use TestBucket or CreateTestBucket instead.
```
Comment (tritone) [inline internal/testutil/storage_cleaner.go:67]: Just FYI, there are a bunch of uses of this outside of GCS samples -- those will have to be replaced as well.

File: internal/testutil/storage_cleaner.go [hunk 2]
```diff
@@ -28,15 +28,43 @@ import (
...
+		t.Fatalf("storage.NewClient: %v", err)
+	}
+	t.Cleanup(func() { client.Close() })
+	return CreateTestBucket(ctx, t, client, projectID, prefix)
+}
+
+// CreateTestBucket creates a new bucket with the given prefix and registers a
+// cleanup function to delete the bucket and any objects it contains.
+// It is equivalent to TestBucket but allows Storage Client re-use.
+func CreateTestBucket(ctx context.Context, t *testing.T, client *storage.Client, projectID, prefix string) string {
```
Comment (tritone) [inline internal/testutil/storage_cleaner.go:48]: Maybe re-name these to indicate what the difference is (e.g. CreateBucket vs CreateBucketWithClient)? Also is it actually important to have both functions available? Creating an extra client in a test is not much overhead.

File: storage/buckets/buckets_test.go [hunk 3]
```diff
@@ -702,8 +702,6 @@ func TestRPO(t *testing.T) {
 		t.Fatalf("createBucketTurboReplication: %v", err)
 	}
 
-	testutil.WaitForBucketToExist(ctx, t, bucket)
```
Comment (tritone) [inline storage/buckets/buckets_test.go:705]: Is this change relevant to this PR?

File: internal/testutil/storage_cleaner.go (hunk 1 above)
Comment (BrennaEpp) [reply internal/testutil/storage_cleaner.go:67]: Yes, I have a follow up ready to replace those.

File: internal/testutil/storage_cleaner.go (hunk 2 above)
Comment (BrennaEpp) [reply internal/testutil/storage_cleaner.go:48]: So I don't want to change the signature of the `CreateTestBucket` function since it will break tests. But I agree that we could just create the extra client instead.

**Review by tritone — APPROVED** [review]
